libloki/api.js

```diff
@@ -1,47 +1,128 @@
-/* global window, textsecure, dcodeIO, StringView, ConversationController */
+/* global window, textsecure, dcodeIO, StringView, ConversationController, _ */
+/* eslint-disable no-bitwise */
 
 // eslint-disable-next-line func-names
 (function() {
   window.libloki = window.libloki || {};
 
-  async function sendBackgroundMessage(pubKey) {
-    return sendOnlineBroadcastMessage(pubKey);
+  const DebugFlagsEnum = {
+    GROUP_SYNC_MESSAGES: 1,
+    CONTACT_SYNC_MESSAGES: 2,
+    AUTO_FRIEND_REQUEST_MESSAGES: 4,
+    SESSION_REQUEST_MESSAGES: 8,
+    SESSION_MESSAGE_SENDING: 16,
+    SESSION_BACKGROUND_MESSAGE: 32,
+    GROUP_REQUEST_INFO: 64,
+    NORMAL_FRIEND_REQUEST_MESSAGES: 128,
+    // If you add anything, be sure it is bitwise safe! (unique and 2 multiples)
+  };
+
+  const debugFlags =
+    DebugFlagsEnum.GROUP_SYNC_MESSAGES |
+    DebugFlagsEnum.CONTACT_SYNC_MESSAGES |
+    DebugFlagsEnum.AUTO_FRIEND_REQUEST_MESSAGES |
+    DebugFlagsEnum.SESSION_REQUEST_MESSAGES |
+    DebugFlagsEnum.SESSION_MESSAGE_SENDING |
+    DebugFlagsEnum.SESSION_BACKGROUND_MESSAGE |
+    DebugFlagsEnum.NORMAL_FRIEND_REQUEST_MESSAGES |
+    DebugFlagsEnum.GROUP_REQUEST_INFO;
+
+  // feel free to change this to window.console.warn to have a stack with all those logs
+  // disable loki logs
+  const debugLogFn = undefined; // window.console.warn
+
+  function logSessionMessageSending(...args) {
+    if (debugFlags | DebugFlagsEnum.SESSION_MESSAGE_SENDING && debugLogFn) {
```

Comment:
```suggestion
    if (debugFlags & DebugFlagsEnum.SESSION_MESSAGE_SENDING && debugLogFn) {
```
All of these checks have to be `&`

Reply:
fixed